plugins/services/src/js/components/modals/NewCreateServiceModalForm.js

```diff
@@ -83,7 +84,22 @@ class NewCreateServiceModalForm extends Component {
     // Hint: When you add something to the state, make sure to update the
     //       shouldComponentUpdate function, since we are trying to reduce
     //       the number of updates as much as possible.
+    // In the Next line we are destructing the config to keep labels as it is and even keep labels with an empty value
+    let { serviceConfig } = ServiceUtil.getServiceJSON(this.props.service);
+    const { labels = {} } = serviceConfig;
 
+    serviceConfig = omit(serviceConfig, ["labels"]);
+
+    let newServiceConfig = Object.assign(
+      {},
+      labels,
+      CreateServiceModalFormUtil.stripEmptyProperties(serviceConfig)
+    );
+    if (Object.keys(labels).length === 0) {
```

Comment:
I think it would be awesome if we could add a test for this behavior, or bring the tests we have for 1.11 to here too. Also if the test is not present on master, I would recommend adding tests on master too (maybe even 1.11)